File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.swift
class Solution {
    func longestOnes(_ nums: [Int], _ k: Int) -> Int {
        var windowStart = 0, mx = 0, zeroCount = 0 
        for windowEnd in 0..<nums.count{
            if nums[windowEnd] == 0{
                zeroCount += 1
            }
            if zeroCount <= k{
                mx = max(mx, windowEnd - windowStart + 1)
            }else{
                while(zeroCount > k){
                    if (nums[windowStart] == 0){
                        zeroCount -= 1
                    }
                    windowStart += 1
                }
            }
        }
        return mx
        
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int zero = 0, one = 0, ans = 0, windowStart = 0, sz = nums.size();
        
//         if(k == 0 ){
            
//         }
        
        for(int windowEnd = 0; windowEnd < sz; windowEnd ++ ){
             int val = nums[windowEnd];
            
             if(val == 0) zero ++ ;
            
            while(windowStart <= windowEnd && zero > k){
                if(nums[windowStart] == 0) zero -- ;
                windowStart ++ ;
            }
            
            ans = max(ans, windowEnd - windowStart + 1);
        }
        
        return ans;
    }
};